chocolatey-diff/Public/Get-FullPackageInfo.ps1

```diff
@@ -19,24 +19,27 @@ function Get-FullPackageInfo {
         [string] $PackageId
     )
 
-    $pkgInfo = @{
-        Id       = $PackageId
-        Url      = "$(Get-ChocolateyPackageGalleryUrl)/${$PackageId}"
-        Versions = @()
-    }
+    if (!(Get-Variable -Name pkgInfo -Scope Script -ErrorAction SilentlyContinue)) {
+
+        $script:pkgInfo = @{
+            Id       = $PackageId
+            Url      = "$(Get-ChocolateyPackageGalleryUrl)/${$PackageId}"
+            Versions = @()
+        }
 
-    $url = "$(Get-ChocolateyRepositoryUrl)/package-versions/${PackageId}?includePreRelease=true"
-    [array]$jsonVersions = Invoke-RestMethod -Uri $url -UseBasicParsing | Select-Object -Unique
-    $apiVersions = $jsonVersions | ForEach-Object { [SemanticVersion]::create($_) } | Sort-Object -Property VersionOnly, Tag -Descending | Select-Object -Unique
+        $url = "$(Get-ChocolateyRepositoryUrl)/package-versions/${PackageId}?includePreRelease=true"
+        [array]$jsonVersions = Invoke-RestMethod -Uri $url -UseBasicParsing | Select-Object -Unique
+        $apiVersions = $jsonVersions | ForEach-Object { [SemanticVersion]::create($_) } | Sort-Object -Property VersionOnly, Tag -Descending | Select-Object -Unique
```

Comment:
```suggestion
        $apiVersions = $jsonVersions |
            ForEach-Object { [SemanticVersion]::Create($_) } |
            Sort-Object -Property VersionOnly, Tag -Descending -Unique
```

Reply:
@vexx32 just FYI.
As the existing implementation is similar to the one in the package request repository, I would just want to say that I tried in that repository to use `Sort-Object -Property VersionOnly, Tag -Descending -Unique`.
Unfortunately, this was something that didn't work as expected there (I do not remember the details, unfortunately), so I think that will have the same problem here.

Reply:
Interesting! I suppose `Sort-Object` would be comparing uniqueness based on the value of the given properties, while `Select-Object` would do value or reference equality on the entire object... I think.

I'd say it's worth testing both here just to be sure, but yes absolutely, if for some reason Select-Object is needed, go for it. 🙂

Reply:
> Interesting! I suppose Sort-Object would be comparing uniqueness based on the value of the given properties, while Select-Object would do value or reference equality on the entire object... I think.

I think it was a bit different, and I think it was the Sort-Object would make a reference comparison, while Select-Object would make a value comparison.
I don't remember exactly, and I just remember it would cause duplicates when not using `Select-Object -Unique`.
But it could of course just have been an error on my side instead, that I fixed later on.

Reply:
I think we can conclude that we are not going to change it.